esp/esp/qsd/tests.py

```diff
@@ -138,7 +138,7 @@ def testPageCorrectness(self):
             qsd_rec_new.save()
             response = self.client.get(self.url)
             self.assertEqual(response.status_code, 200)
-            self.assertTrue('Testing 456' in response.content)
+            self.assertTrue('Testing 456' in str(response.content))
```

Comment:
In python 3, `str` doesn't convert from bytes to string in the way you think it would.  For example:

```
>>> u = b'xyz'
>>> print(str(u))
b'xyz'
>>> len(str(u))
6
```
The best thing would be to figure out some way to make `response.content` return a string rather than bytes in the first place so you wouldn't have to do this conversion in the first place. If there's no way to do that, then you can do `response.content.decode('UTF-8')` to do the conversion.

Reply:
More generally: any time you convert from string to bytes or vica versa, you have to specify an encoding.

Reply:
Looks like the `response.content` is always bytes: https://docs.djangoproject.com/en/3.2/topics/testing/tools/#django.test.Response.content.